messenger.rst

```diff
@@ -694,128 +922,252 @@ Defining such configurable middleware is based on Symfony's
+                    async_priority_normal: # ...
+                    image_transport: # ...
+
+                routing:
+                    # ...
+                    'App\Message\UploadedImage': [image_transport, async_priority_normal]
 
     .. code-block:: xml
 
-        <!-- config/services.xml -->
+        <!-- config/packages/messenger.xml -->
         <?xml version="1.0" encoding="UTF-8" ?>
         <container xmlns="http://symfony.com/schema/dic/services"
             xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+            xmlns:framework="http://symfony.com/schema/dic/symfony"
             xsi:schemaLocation="http://symfony.com/schema/dic/services
-                https://symfony.com/schema/dic/services/services-1.0.xsd">
+                https://symfony.com/schema/dic/services/services-1.0.xsd
+                http://symfony.com/schema/dic/symfony
+                https://symfony.com/schema/dic/symfony/symfony-1.0.xsd">
 
-            <services>
-                <service id="Your\Transport\YourTransportFactory">
-                   <tag name="messenger.transport_factory"/>
-                </service>
-            </services>
+            <framework:config>
+                <framework:messenger>
+                    <framework:transport name="async_priority_normal" dsn="..."/>
+                    <framework:transport name="image_transport" dsn="..."/>
+
+                    <framework:routing message-class="App\Message\UploadedImage">
+                        <framework:sender service="image_transport"/>
+                        <framework:sender service="async_priority_normal"/>
+                    </framework:routing>
+                </framework:messenger>
+            </framework:config>
         </container>
 
     .. code-block:: php
 
-        // config/services.php
-        use Your\Transport\YourTransportFactory;
+        // config/packages/messenger.php
+        $container->loadFromExtension('framework', [
+            'messenger' => [
+                'transports' => [
+                    'async_priority_normal' => '...',
+                    'image_transport' => '...',
+                ],
+                'routing' => [
+                    'App\Message\UploadedImage' => ['image_transport', 'async_priority_normal']
+                ]
+            ],
+        ]);
+
+That's it! You can now consume each transport:
+
+.. code-block:: terminal
+
+    # will only call ThumbnailUploadedImageHandler when handling the message
+    $ php bin/console messenger:consume image_transport -vv
+
+    $ php bin/console messenger:consume async_priority_normal -vv
+
+.. caution::
+
+    If a handler does *not* have ``from_transport`` config, it will be executed
+    on *every* transport that the message is received from.
 
-        $container->register(YourTransportFactory::class)
-            ->setTags(['messenger.transport_factory']);
+Extending Messenger
+-------------------
 
-Use your Transport
+Envelopes & Stamps
```

Comment:
It would be awesome an another example with the SerializerStamp 😄

Reply:
Good catch. I added it to the serializer section. But not down here because, again, I think that most people won't use the Symfony serializer.